src/scheduler/SchedulerSubSystem.java

```diff
@@ -37,26 +45,32 @@ public void schedule() {
 
 
 		if (m instanceof ElevatorRequestMessage) {
+			Long initTime = System.nanoTime();
 			threadPool.execute(new Runnable() {
 				@Override
 				public void run() {
 					handleElevatorRequest((ElevatorRequestMessage) m);
 				}
 			});
+			ElevatorRequestTimes.add(initTime - System.nanoTime());
```

Comment:
These should get executed RIGHT after spinning up the thread which would just be the time it took to spin up the thread. Instead, we should run the add in the `run()` with a lock on the list.

Reply:
Ur right I just wasn't thinking, also note that I just made the list object synchronized so we don't have to explicitly lock it every time